packages/core/strapi/lib/commands/utils/commander.js

```diff
@@ -9,15 +14,60 @@ const parseInputList = (value) => {
   return value.split(',');
 };
 
+const parseURL = (value) => {
+  try {
+    const url = new URL(value);
+    if (!url.host) {
+      throw new InvalidOptionArgumentError(`Could not parse url ${value}`);
+    }
+
+    return url;
+  } catch (e) {
+    throw new InvalidOptionArgumentError(`Could not parse url ${value}`);
+  }
+};
+
+/**
+ * assert that a URL object has a protocol value
+ *
+ * @param {URL} url
+ * @param {string[]|undefined} protocol
```

Comment:
Based on what the function is doing, I would suggest using this signature instead
```suggestion
 * @param {string[]|string} [protocol]
```